File: src/electron-app/scanner.js
/* ============================================================
   scanner.js — Hermes Agent 安装目录扫描
   ============================================================ */

const path = require('path')
const fs = require('fs')
const os = require('os')

function scanHermesInstall(app) {
  const candidates = []

  candidates.push(process.cwd())

  const home = os.homedir()
  const localAppData = process.env.LOCALAPPDATA || ''

  candidates.push(path.join(home, '.hermes', 'hermes-agent'))
  candidates.push(path.join(home, 'hermes-agent'))
  if (localAppData) {
    candidates.push(path.join(localAppData, 'hermes'))
    candidates.push(path.join(localAppData, 'hermes', 'hermes-agent'))
  }

  let dir = app.isPackaged ? path.dirname(app.getPath('exe')) : path.resolve(__dirname, '..', '..', '..')
  for (let i = 0; i < 6; i++) {
    candidates.push(path.join(dir, 'hermes-agent'))
    candidates.push(dir)
    const parent = path.dirname(dir)
    if (parent === dir) break
    dir = parent
  }

  candidates.push('D:\\hermes-agent')
  candidates.push('D:\\hermes')
  candidates.push('D:\\hermes-offline')

  const found = []
  for (const c of candidates) {
    try {
      const p = path.resolve(c)
      if (fs.existsSync(path.join(p, 'pyproject.toml'))) {
        if (!found.includes(p)) found.push(p)
      }
      const sub = path.join(p, 'hermes-agent')
      if (fs.existsSync(path.join(sub, 'pyproject.toml'))) {
        if (!found.includes(sub)) found.push(sub)
      }
    } catch {}
  }

  return found
}

module.exports = { scanHermesInstall }

File: src/scanner.py
"""Hermes Agent 安装目录扫描（使用公共自动发现）"""

import os
from pathlib import Path

# 优先使用 installer.py 里的 find_hermes_installation
try:
    from installer import find_hermes_installation
    _use_installer = True
except ImportError:
    _use_installer = False


def scan_hermes_targets():
    """扫描可能的 Hermes Agent 安装目录，返回路径列表"""
    found = []

    # 使用 installer.py 的自动发现
    if _use_installer:
        try:
            path, desc = find_hermes_installation()
            if path:
                found.append(str(path))
        except Exception:
            pass

    # 补充旧版扫描逻辑
    local_app_data = os.environ.get("LOCALAPPDATA", "")
    home = Path.home()
    fallback = [
        Path(local_app_data) / "hermes" / "hermes-agent" if local_app_data else None,
        Path(local_app_data) / "hermes" if local_app_data else None,
        home / "hermes-agent",
        home / "hermes" / "hermes-agent",
        Path("D:\\hermes-agent"),
        Path("D:\\hermes"),
    ]

    for p in fallback:
        if not p or not p.exists():
            continue
        s = str(p)
        if s not in found and (p / "pyproject.toml").exists():
            found.append(s)
        elif s not in found:
            sub = p / "hermes-agent"
            if sub.exists() and (sub / "pyproject.toml").exists():
                found.append(str(sub))

    return found